network/network.go

```diff
@@ -46,6 +46,8 @@ type Network interface {
 	Connect() error
 	// Nodes returns list of network nodes
 	Nodes() []Node
+	// Topology returns node topology up to MaxDepth hops
+	Topology() Node
```

Comment:
Does this not seem to basically return Node which is the existing Node embedded in this Network? Should we maybe just return the entire topology in Peers and remove the Topology() method?

Reply:
Yes and No. 

Yes it does return the network `Node`
No, it returns the network `Node` [graph] that contains all the reachable nodes up to 3 hops.

Reply:
Lets return the known topology (max depth 3) here in Peers() and remove Nodes()/Topology() methods in the interface.

Reply:
I see what you mean here, but won't this overload the term `Peers()`?

Reply:
It depends on the expectation. If Peers() returns a list of nodes and each node has a list of Peers then you probably expect that you can continue to walk that as far as it goes. I just think limiting the surface area of the API early on is better because its going to be harder to claw it back later.

Reply:
You expect but at the moment `Peers()` returns literary just a list of immediate neighbours like we originally agreed. Not a graph. A list of immediate neighbours. Nothing else. If you want to replace the `Peers()` logic with what the `Topology()` does now and then scrap the `Topology` then we can do that, but then I feel we're overloading term `Peers` or something. Im ok to do this if you want me to.